compiler/codegen/cg-type.cpp

```diff
@@ -187,6 +205,217 @@ void EnumType::codegenDef() {
+                              uint64_t& structSize, int& structAlignment)
+{
+  GenInfo* info = gGenInfo;
+  int      paramID = 0;
+  uint64_t currSize = 0;
+  int      currAlignment = ALIGNMENT_DEFER;
+  INT_ASSERT(GEPMap.empty());  // we clear it if explicit padding is needed
+  llvm::Type* padTy =
+    explicitPadding ? llvm::Type::getInt8Ty(gContext->llvmContext()) : nullptr;
+
+  // Performs updates when explicitPadding==false.
+  // This avoids the need to compute fieldSize and fieldAlign.
+  auto addFieldWhenNotEP = [&](const char* fieldName, llvm::Type* fieldType) {
+    GEPMap.insert(std::pair<std::string, int>(fieldName, paramID));
+    paramID++;
+    params.push_back(fieldType);
+  };
+
+  // Performs updates when explicitPadding==true.
+  // Assumes that padding has already been inserted as needed.
+  auto addFieldWhenEP = [&](const char* fieldName, llvm::Type* fieldType,
+                            uint64_t fieldSize, int fieldAlign) {
+    GEPMap.insert(std::pair<std::string, int>(fieldName, paramID));
+    paramID++;
+    params.push_back(fieldType);
+    aligns.push_back(fieldAlign);
+    currSize += fieldSize;
+    currAlignment = std::max(currAlignment, fieldAlign);
+    if (fieldAlign > 16 && ag->aggregateTag == AGGREGATE_CLASS) {
+      USR_FATAL_CONT(ag, "alignments > 16 are currently not implemented"
+                         " for classes");
+      USR_PRINT(ag, "alignment %d is requested by field '%s' of class '%s'",
+                fieldAlign, fieldName, ag->symbol->name);
+    }
+  };
+
+  // Inserts 'numBytes' of padding when explicitPadding==true.
+  auto insertPadding = [&](int numBytes) {
+    // GEPMap is unaffected
+    paramID++;
+    params.push_back(llvm::ArrayType::get(padTy, numBytes));
+    aligns.push_back(1);
+    currSize += numBytes;
+    // currAlignment is unaffected
+  };
+
+
+  if (ag->aggregateTag == AGGREGATE_UNION) {
+    //// a union ////
+    const char* uidName = "_uid";
+    llvm::Type* uidType = llvm::Type::getInt64Ty(info->module->getContext());
+    if (explicitPadding)
+      addFieldWhenEP(uidName, uidType, 8,
+                     (int)DL.getABITypeAlign(uidType).value());
+    else
+      addFieldWhenNotEP(uidName, uidType);
+
+    llvm::Type *largestType = NULL;
+    uint64_t largestSize = 0;
+
+    for_fields(field, ag) {
+      llvm::Type* fieldType = field->type->symbol->getLLVMType();
+      INT_ASSERT(fieldType);
+      uint64_t fieldSize = DL.getTypeStoreSize(fieldType);
+
+      if(fieldSize > largestSize) {
+        largestType = fieldType;
+        largestSize = fieldSize;
+      }
+
+      GEPMap.insert(std::pair<std::string, int>(field->cname, paramID));
+      GEPMap.insert(std::pair<std::string, int>(
+            std::string("_u.") + field->cname, paramID));
+
+      // TODO account for field's alignment in case it is >8
+    }
+
+    llvm::StructType * st;
+    // handle an empty union.
+    if( largestType ) {
```

Comment:
nit, spacing on condition

Reply:
Both the above conditionals are part of code that was moved from elsewhere in this file. I propose to retain the original spacing, although it is a weak preference.

Reply:
I think if you are touching the code anyways, you might as well leave it in a better state